src/Microsoft.Extensions.Caching.Memory/MemoryCache.cs

```diff
@@ -37,7 +40,7 @@ public MemoryCache(IOptions<MemoryCacheOptions> optionsAccessor)
             var options = optionsAccessor.Value;
             _entries = new Dictionary<object, CacheEntry>();
             _entryLock = new ReaderWriterLockSlim();
-            _entryExpirationNotification = EntryExpired;
```

Comment:
Why was this removed? I'm told we have to cache these explicitly because otherwise the compiler allocates them every call.